### PR TITLE
Feature/admin

### LINKED ISSUE #165: Create ADMIN page to CREATE, READ, UPDATE, and DELETE Products into MongoDB using existing CRUD API
# Issue Report

## Title

Create ADMIN page to CREATE, READ, UPDATE, and DELETE Products into MongoDB using existing CRUD API

## Description
This issue involves building an Admin page that allows for the Create, Read, Update, and Delete (CRUD) operations on Products stored in MongoDB. The page will interact with an existing CRUD API for product management. The page should include a form for product creation and editing, a table to display all products, and functionality to delete products.

## Steps to Reproduce the Issue
Action 1: Set up the admin page with components such as a form for product creation and a table for displaying products.
Action 2: Implement the connection to the existing CRUD API for fetching, creating, updating, and deleting products in MongoDB.
Action 3: Add functionality to handle user inputs for creating new products and updating existing ones.
Action 4: Test the form and table components to ensure proper data flow between the frontend and backend.

## Expected Behavior

Products should be created, displayed, updated, and deleted through the admin page.
The admin should be able to perform these operations by interacting with the form and table, and changes should reflect in the MongoDB database.
Data validation errors should be handled properly.

---

Thank you for taking the time to report this issue! We appreciate your help in improving our project.

### PR DESCRIPTION
**Pull request by samriddhitiwary**
https://github.com/user-attachments/assets/ffca417e-1792-4f84-b73a-8d79731260e5

# Pull Request Template

**Title:** Create ADMIN page to CREATE, READ, UPDATE, and DELETE Products into MongoDB using existing CRUD API

## Description

Please provide a clear and concise description of what this pull request does, including any relevant details.

## Related Issue

Link to the issue that this pull request addresses (if applicable). Use the format: `Fixes #165 `

## Changes Made

This issue involves building an Admin page that allows for the Create, Read, Update, and Delete (CRUD) operations on Products stored in MongoDB. The page will interact with an existing CRUD API for product management. The page should include a form for product creation and editing, a table to display all products, and functionality to delete products.

## Checklist

- [x] I have tested my code and it works as expected.
- [x] I have added/updated documentation where necessary.
- [x] I have included tests for new features.
- [x] I have followed the project's coding standards.

- **Checklist:**
  - [x] I have tested my code and it works as expected.
  - [ ] I have added documentation for the changes made.
  - [x] I have included tests for the registration form validation.

### CODE REVIEW

**Comment by saurabhbakolia** [comment]
Hey @samriddhitiwary , 
This is cool, I am reviewing it.

**Comment by samriddhitiwary** [comment]
@thakuratul2 I have resolved the conflicts

**Comment by samriddhitiwary** [comment]
@thakuratul2 I have resolved the conflict, could you please review the PR

**Review by saurabhbakolia — CHANGES_REQUESTED** [review]

**Comment by samriddhitiwary** [comment]
@saurabhbakolia Ok I will check it again

File: backend/server.js [hunk 1]
```diff
@@ -1,44 +1,51 @@
...
-); // `extended` option allows for rich objects and arrays to be encoded into the URL-encoded format
+); // extended option allows for rich objects and arrays to be encoded into the URL-encoded format
 app.use(cookieParser());
 app.use(helmet());
 
 app.use("/api/auth", authRoutes); // Authentication routes
 app.use("/api/cart", authMiddleware, cartRoute); // Cart routes
 app.use("/api/product", productRoute); // Product routes
 app.use("/api/user", userRoute); // user routes
-app.use("/api/auth", authStatusRoute); // Check Auth Status
```
[the remark this answers is not in the document]
Comment (samriddhitiwary) [reply backend/server.js:41]: @saurabhbakolia 
I am sorry, but, for time being, can hold all Auth related flow, Its causing lots of issue in local. We can have separate issue to fix all the auth related work. 

image

File: backend/api/routes/productRoute.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const authMiddleware = require('../middleware/authMiddleware');
 const productController = require('../controllers/productController');
 
 // Route to add a new product (only for admins)
-router.post('/add', authMiddleware, productController.addProduct);
```
[the remark this answers is not in the document]
Comment (samriddhitiwary) [reply backend/api/routes/productRoute.js:7]: As of now, I have disabled all the auth flow. Later, we can add it in separate issue and PR?

File: src/components/UpdateProduct.jsx [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+      }
+
+      .btn-primary {
+        width: 50%;
+      }
+    }
+  `}
+</style>
+
+      <div className='container mt-5'>
```
[the remark this answers is not in the document]
Comment (samriddhitiwary) [reply src/components/UpdateProduct.jsx:151]: Sure, I am making these changes.

File: backend/server.js [hunk 4]
```diff
@@ -48,11 +55,11 @@ app.use("/api/auth", authStatusRoute); // Check Auth Status
 // });
 // app.use(limiter);
 
-const PORT = process.env.PORT || 8080;
-if (process.env.NODE_ENV !== 'test')
-  app.listen(PORT, () => {
-    console.log(`Server running on port ${PORT}`);
-    console.log(`API available at http://localhost:${PORT}/api/auth`);
-  });
+// const PORT = process.env.PORT || 8080;
```
[the remark this answers is not in the document]
Comment (samriddhitiwary) [reply backend/server.js:58]: I am trying to connect the MongoDB ATLAS. It seems, existing code was not compatible with ATLAS, So when i modified, It started working. I did not try with local mongo db.

File: src/App.js [hunk 5]
```diff
@@ -10,6 +10,10 @@ import ProtectedRoute from './utils/ProtectedRoute';
 import ContributorsPage from './pages/ContributorsPage';
 import { lazy, Suspense } from 'react';
 import ContactUsPage from './pages/ContactUsPage';
+import Admin from './components/Admin';
```
[the remark this answers is not in the document]
Comment (samriddhitiwary) [reply src/App.js:13]: Sure, I will do it. I missed it.

File: package.json [hunk 6]
```diff
@@ -13,11 +13,13 @@
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
     "axios": "^1.6.0",
+    "bootstrap": "^5.3.3",
```
[the remark this answers is not in the document]
Comment (samriddhitiwary) [reply package.json:16]: I will check, I am not using anywhere. I will remove it.

File: package.json [hunk 7]
```diff
@@ -13,11 +13,13 @@
...
     "firebase": "^10.14.0",
     "framer-motion": "^11.9.0",
     "myapp": "file:",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-hot-toast": "^2.4.1",
```
[the remark this answers is not in the document]
Comment (samriddhitiwary) [reply package.json:22]: I will check and will use chakra-ui.

File: src/components/AddNewProduct.jsx [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+  return (
+    <>
+      <Announcement />
+      <Navbar />
+      <div
+        style={{
+          background: `linear-gradient(rgba(255, 255, 255, 0.5), rgba(255, 255, 255, 0.5)), url('https://images.pexels.com/photos/6984650/pexels-photo-6984650.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940') center`,
+          backgroundSize: 'cover',
+          backgroundAttachment: 'fixed', // Keep the background fixed during scroll
+          minHeight: '100vh', // Ensure it covers the full height of the viewport
```
[the remark this answers is not in the document]
Comment (samriddhitiwary) [reply src/components/AddNewProduct.jsx:72]: Sure, will do it.

**Comment by saurabhbakolia** [comment]
> @saurabhbakolia Ok I will check it again

Hey @samriddhitiwary,

if you can make the required changes so that I can merge this PR.

**Comment by samriddhitiwary** [comment]
@saurabhbakolia Yes yes

**Comment by samriddhitiwary** [comment]
@saurabhbakolia I will create another fresh PR and I will close this PR

**Pull request closed without merging**